http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java

```diff
@@ -427,9 +431,19 @@ private static ServletContextHandler createServletContext(
         context.addServlet(servletHolder, "/*");
 
         for (Map.Entry<String, Servlet> servlet : servlets.entrySet()) {
-            ServletHolder holder = new ServletHolder(servlet.getValue());
-            holder.setInitParameters(ImmutableMap.copyOf(parameters));
-            context.addServlet(holder, servlet.getKey());
+            if (config.isAuthEnabled()) {
+                SecureServlet secureServlet = new SecureServlet(servlet.getValue(),
+                        authorizer,
+                        getRolesFromClassMetadata(servlet.getValue()),
+                        config);
+                ServletHolder holder = new ServletHolder(secureServlet);
```

Comment:
the parameters should be set for this servlet as well, similar as for non sercure servlet. Also make sure if parameters are delegated by the `SecureServlet` to the underlying servlet.

Reply:
In order to access `initParameters` from delegate servlet, the `servletConfig` has to be initialized.

See: https://github.com/prestodb/airlift/pull/17/files#diff-47e526a2479bc0cbe59875706a697658R62-R68